docs/source/developer-documentation/development-tutorial/creating-boefje.rst

```diff
@@ -0,0 +1,390 @@
+
+.. code:: json
+
+   {
+     "status": "COMPLETED",
+     "files": [
+        {
+            "name": "my-first-output",
+            "content": "YWJjCg==",
+            "tags": ["boefje/dns-records"]
+        },
+        {
+            "name": "my-second-output",
+            "content": "ZGVmCg==",
+            "tags": ["application/json", "boefje/dns-records", "other/mime-types"]
+        }
+    ]
+   }
+
+Such an image can be added through the *Add Boefje* button in the KATalogus.
+
+Creating an OCI image using our Base Image
+------------------------------------------
+
+Because a lot of our boefjes follow a similar pattern, we have created a Python base image you can use to quickly
+create a Python boefje. This base image handles communicating with the boefje API and lets you create boefjes as we've
+done in the past.
+
+Creating a new boefje directory
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Inside ``boefjes/boefjes/plugins/`` create a new folder with a name
+starting with ``kat_``. In this example we use ``kat_hello_katty``.
+
+.. warning::
+    Of course, you could decide to create these files elsewhere and still build your image using our base image. However,
+    adding it to this folder makes sure the new boefje is imported in the KAT-alogus automatically. Else you will have
+    to add the boefje using the ``Add Boefje`` button in the KAT-alogus and specify your newly created image tag manually.
+
+Inside this folder we need to have the following files:
+
+.. code:: shell
+
+   $ tree boefjes/boefjes/plugins/kat_hello_katty/
+   ├── __init__.py
+   ├── boefje.json
+   ├── cover.jpg
+   ├── description.md
+   ├── boefje.Dockerfile
+   ├── main.py
+   └── schema.json
+
+``__init__.py``
+~~~~~~~~~~~~~~~
+
+This file stays empty.
+
+``boefje.json``
+~~~~~~~~~~~~~~~
+
+This file contains information about our boefje. For example, this file
+contains information about what OOIs our boefje should be looking out
+for. Here is the example we will be using:
+
+.. code:: json
+
+   {
+     "id": "hello-katty",
+     "name": "Hello Katty",
+     "description": "A simple boefje that can say hello",
+     "consumes": ["IPAddressV4", "IPAddressV6"],
+     "scan_level": 0,
+     "oci_image": "openkat/hello-katty"
+   }
+
+-  **``id``**: A unique identifier for the boefje.
+-  **``name``**: A name to display in the KAT-alogus.
+-  **``description``**: A description in the KAT-alogus.
+-  **``consumes``**: A list of OOI types that trigger the boefje to run.
+   Whenever one of these OOIs gets added, this boefje will run with that
+   OOI. In our case, we will run our boefje whenever a new IPAddressV4
+   or IPAddressV6 gets added.
+-  **``scan_level``**: A scan level that decides how intrusively this
+   boefje will scan the provided OOIs. Since we will not make any
+   external requests our boefje will have a scan level of 0.
+-  **``oci_image``**: The name of the docker image that is provided
+   inside ``boefjes/Makefile``
```

Comment:
```suggestion
-  **id**: A unique identifier for the boefje.
-  **name**: A name to display in the KAT-alogus.
-  **description**: A description in the KAT-alogus.
-  **consumes**: A list of OOI types that trigger the boefje to run.
   Whenever one of these OOIs gets added, this boefje will run with that
   OOI. In our case, we will run our boefje whenever a new IPAddressV4
   or IPAddressV6 gets added.
-  **scan_level**: A scan level that decides how intrusively this
   boefje will scan the provided OOIs. Since we will not make any
   external requests our boefje will have a scan level of 0.
-  **oci_image**: The name of the docker image that is provided
   inside ``boefjes/Makefile``
```